### PR TITLE
Python: Replace SQLContext with SparkSession

### PR DESCRIPTION
**Pull request by ttc0419**
According to the [official document](https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/SQLContext.html), ```SQLContext``` is deprecated since spark 2.0.  Use ```SparkSession``` instead.

**Notes:**
This MR also removes CI for spark 2.4.8 since it does not support ```SparkSession.getActiveSession()```.  And according to [versioning policy](https://spark.apache.org/versioning-policy.html), spark 2.4.x is EOL and the last version was released in 5 years ago.  There's little meaning to support it.

Closes: #423

### CODE REVIEW

**Comment by ttc0419** [comment]
Tagging @WeichenXu123, since you reviewed the most recent MR.

**Comment by WeichenXu123** [comment]
@ttc0419 Thank you! I will take a look soon.

File: python/graphframes/graphframe.py
Comment (WeichenXu123) [inline python/graphframes/graphframe.py:66]:
```suggestion
        self._spark = SparkSession.getActiveSession()
```

Comment (WeichenXu123) [reply python/graphframes/graphframe.py:66]: `ss` var name is not clear.

**Comment by WeichenXu123** [comment]
The trait `GraphFrameTestSparkContext` still has code `protected override def _sqlContext: SQLContext = self.sqlContext`, can we replace it with `SparkSession` ?

**Comment by ttc0419** [comment]
> The trait `GraphFrameTestSparkContext` still has code `protected override def _sqlContext: SQLContext = self.sqlContext`, can we replace it with `SparkSession` ?

I'm not familiar with Scala, probably need someone else and another MR to do that.

File: python/graphframes/graphframe.py
Comment (ttc0419) [reply python/graphframes/graphframe.py]: Done.

**Comment by WeichenXu123** [comment]
> > The trait `GraphFrameTestSparkContext` still has code `protected override def _sqlContext: SQLContext = self.sqlContext`, can we replace it with `SparkSession` ?
> 
> I'm not familiar with Scala, probably need someone else and another MR to do that.

We can do it in follow-up PR.

**Comment by ttc0419** [comment]
@WeichenXu123 Anything else need to be done before merging this?

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Comment by ttc0419** [comment]
@WeichenXu123 Do you need another reviewer to merge this?

**Comment by WeichenXu123** [comment]
> @WeichenXu123 Do we need another reviewer to merge this?

I can merge it.

**Pull request merged**
